nni/retiarii/converter/graph_gen.py

```diff
@@ -243,63 +275,162 @@ def handle_single_node(node):
+            new_node = ir_graph.add_node(build_full_name(module_name, Type.Attr, global_seq),
+                                            node_type, attrs)
+            node_index[node] = new_node
         elif node.kind() == 'prim::If':
             last_block_node = handle_if_node(node)
             node_index[node] = last_block_node
-        elif node.kind() == 'prim::GetAttr':
-            pass
+        elif node.kind() == 'prim::Loop':
+            print('mygraph: ', sm_graph)
+            raise RuntimeError('Loop has not been supported yet!')
         else:
             raise RuntimeError('Unsupported kind: {}'.format(node.kind()))
 
-        if node in node_index:
-            return node_index[node]
-        else:
-            return None
+        return node_index[node]
 
     for node in sm_graph.nodes():
         handle_single_node(node)
 
     return node_index
 
+def merge_aten_slices(ir_graph):
+    """
+    if there is aten::slice node, merge the consecutive ones together
+    """
+    head_slice_nodes = []
+    has_slice_node = False
+    for node in ir_graph.hidden_nodes:
+        if node.operation.type == 'aten::slice':
+            has_slice_node = True
+            for pred in node.predecessors:
+                if pred.operation.type not in ['aten::slice', 'prim::Constant']:
+                    head_slice_nodes.append(node)
+                    break
+    if has_slice_node:
+        assert head_slice_nodes
+    
+    for head_node in head_slice_nodes:
+        slot = 0
+        new_slice_node = ir_graph.add_node(build_full_name(head_node.name, 'merged'), Type.MergedSlice)
+        assert len(head_node.incoming_edges) == 5
+        for edge in head_node.incoming_edges:
+            edge.tail = new_slice_node
+        slot += 5
+        node = head_node
+        while len(node.successors) == 1 and node.successors[0].operation.type == 'aten::slice':
+            suc_node = node.successors[0]
+            assert len(suc_node.incoming_edges) == 5
+            for edge in suc_node.incoming_edges:
+                if edge.tail_slot == 0:
+                    edge.remove()
+                else:
+                    edge.tail = new_slice_node
+                    edge.tail_slot = slot + edge.tail_slot - 1
+            slot += 4
+            ir_graph.hidden_nodes.remove(node)
+            node = suc_node
+        
+        for edge in node.outgoing_edges:
+            edge.head = new_slice_node
+        ir_graph.hidden_nodes.remove(node)
+    
+
+def refine_graph(ir_graph):
+    """
+    Do the following process to simplify graph:
+    1. remove unconnected constant node
+    2. remove unconnected getattr node
+    3. FIXME: merge slice
+    """
+    # some constant is not used, for example, function name as prim::Constant
+    remove_unconnected_nodes(ir_graph, targeted_type='prim::Constant')
+    remove_unconnected_nodes(ir_graph, targeted_type='prim::GetAttr')
+    merge_aten_slices(ir_graph)
+
+def _handle_layerchoice(module):
```

Comment:
So you finally decided not to make `LayerChoice` placeholder?

Reply:
let's refactor this part in the future